src/marshmallow/class_registry.py

```diff
@@ -15,8 +16,11 @@
 # }
 _registry = {}
 
+if typing.TYPE_CHECKING:
```

Comment:
Honest question: Is this the accepted way to do things when an import is only used for typing? In practice there should be near-zero cost for this import, but if this is the accepted wisdom, then I guess it's fine.

Reply:
I'm totally with you on micro-optimizing import time....

So this was a rather instinctual attempt to prevent circular imports. At a quick glance I'm not sure whether this is an actual issue or not for `class_registry.py` (still working on my local test environment! Maybe that'll help answer the question)

I'll take it out if it's not necessary

Reply:
In any case, it certainly is an eyesore. If only the python devs could figure out how to make the whole circular import problem just go away for those of us who aren't really using python as a scripting language

Reply:
Looks like my gut was right on this one! Circular import city:

```python
ImportError while loading conftest '/home/fuhrysteve/projects/marshmallow/tests/conftest.py'.
tests/conftest.py:4: in <module>
    from tests.base import User, UserSchema, Blog
tests/base.py:9: in <module>
    from marshmallow import Schema, fields, post_load, validate, missing
.tox/py36/lib/python3.6/site-packages/marshmallow/__init__.py:1: in <module>
    from marshmallow.schema import Schema, SchemaOpts
.tox/py36/lib/python3.6/site-packages/marshmallow/schema.py:13: in <module>
    from marshmallow import base, fields as ma_fields, class_registry
.tox/py36/lib/python3.6/site-packages/marshmallow/fields.py:13: in <module>
    from marshmallow import validate, utils, class_registry
.tox/py36/lib/python3.6/site-packages/marshmallow/class_registry.py:12: in <module>
    from marshmallow import Schema
E   ImportError: cannot import name 'Schema'
```